examples/ui-demo/src/components/modals/Erc20/Erc20Modal.tsx

```diff
@@ -0,0 +1,398 @@
+    toastText: "NFT minted successfully",
+  });
+
+  const {
+    estimateGasAsync: estimateMintNftFee,
+    isEstimating: isEstimatingGas,
+  } = useEstimateGasErc20Sponsorship({
+    clientOptions: { mode: accountMode, chain, transport },
+  });
+
+  const balanceFloat = parseFloat(balance ?? "0");
+  const readyToBuyNft = balanceFloat >= MINIMUM_USDC_BALANCE;
+
+  const numericBalance = Math.floor(balanceFloat * 100) / 100;
+  const balanceDisplay = isLoadingBalance
+    ? "Loading..."
+    : isErrorBalance
+      ? "Error"
+      : `${numericBalance === 0 ? "0" : numericBalance.toFixed(2)} USDC`;
+
+  const handleGetUSDC = async () => {
+    if (!accountAddress) return;
+
+    try {
+      await mintAsync();
+      await refetchBalance();
+    } catch (e) {
+      console.error("Failed to get USDC:", e);
+    }
+  };
+
+  const handleClose = () => {
+    onClose();
+  };
+
+  const handleBuyNFT = async () => {
+    if (!accountAddress) return;
+
+    if (mintNftTxHash) {
+      resetMintNft();
+      return;
+    }
+
+    try {
+      const uos = await getNftMintBatchUOs(accountAddress);
+      await mintNftAsync(uos);
+      await refetchBalance();
+      onNftMinted?.();
+    } catch (e) {
+      console.error("Failed to buy NFT:", e);
+    }
+  };
+
+  useEffect(() => {
+    if (isOpen && accountAddress) {
+      refetchBalance();
+    }
+  }, [isOpen, accountAddress, refetchBalance]);
+
+  useEffect(() => {
+    const fetchAndEstimateFee = async () => {
+      if (
+        isOpen &&
+        accountAddress &&
+        ethPriceData?.price &&
+        !isLoadingEthPrice
+      ) {
+        try {
+          const uos = await getNftMintBatchUOs(accountAddress);
+          const feeResult = await estimateMintNftFee(uos);
+          if (feeResult && feeResult.feeEth) {
+            const calculatedNetworkFee =
+              parseFloat(feeResult.feeEth) * ethPriceData.price;
+            setNetworkFee(calculatedNetworkFee);
+          } else {
+            console.warn("Fee estimation did not return expected data.");
+            setNetworkFee(0);
+          }
+        } catch (error) {
+          setNetworkFee(0);
+        }
+      } else if (isOpen && accountAddress && !isLoadingEthPrice) {
+        setNetworkFee(0);
+        if (!ethPriceData?.price) {
+          console.warn("ETH price not available for fee estimation.");
+        }
+      }
+    };
+
+    fetchAndEstimateFee();
+  }, [
+    isOpen,
+    accountAddress,
+    estimateMintNftFee,
+    ethPriceData?.price,
+    isLoadingEthPrice,
+    balance,
+  ]);
+
+  useEffect(() => {
+    resetMintNft();
+    resetMint();
+  }, [accountMode, resetMintNft, resetMint]);
+
+  const buyNftButtonEnabled =
+    readyToBuyNft && !isMintingNft && !isLoadingNftClient;
+
+  let bagIconColor = mintNftTxHash
+    ? "#475569"
+    : buyNftButtonEnabled
+      ? "white"
+      : "#CBD5E1";
+
+  return (
+    <Dialog isOpen={isOpen} onClose={handleClose}>
```

Comment:
if it's not too much trouble, should the erc-20 minting modal be centered b/w ~`770px` & `1024px` viewport? it looks oddly to the right.

Reply:
if it's a pain to fix, maybe we can worry about it as a fast-follow to not block the launch of this.